[IMP] Livechat: chatbots 17.0 (#8306)
odoo/documentation: content/applications/websites/livechat/chatbots.rst

would.

Chatbots can be customized to fill various roles, from customer support, to creating leads, to

> collecting contact information. The goal of each chatbot can vary based on several criteria,

Reviewer: This might be beyond silly, but I just realized when testing a chatbot, the chatbot works in which apps? In testing, it sends me to a website- but it can also work just in the database- and I assume the portal? I honestly don't know- and just thought of it- so that could be mentioned somewhere here the various places it can- and cannot- appear? Just an idea.

Author: Chatbots can be added to any livechat channel, so they can appear on any web page where a channel is added.

Chatbot scripts
---------------

After the new chatbot has been created and named, the next step is to create a script. Chatbot
conversations follow an accompanying script. These scripts are comprised of lines of dialog, each
designed to deliver or capture information.

To create a chatbot script, click :guilabel:`Add a Line` in the :guilabel:`Script` tab of the

> chatbot detail page, and a :guilabel:`Create Script Steps` pop-up form appears.

Reviewer 2: This UI element is a _modal_ :)
```suggestion
chatbot detail page, and a :guilabel:`Create Script Steps` modal appears.

```

This step prompts visitors to provide their email address, which is stored and can be used by team
members later to follow up with additional information.

> The **only** accepted inputs for this step type are email addresses that are in a valid format. If a

Reviewer: Just adding this here since I can't add it in the rest of this email section. You might want to mention that the question should be related to this- maybe in an example block? I was confused for a second, the realized the question has to be "Can you provide me with your email address so I can reach you?'or something like that. 

I was also going to recommend an example be added for each of these Step Types to fully illustrate the possibilities. But that might be overkill- however, it could help someone as they are trying to think of how to format their chatbot. Not necessary- just an idea!

Author: I agree examples for each would be overkill, but I think since the image with this section also has the question and "invalid email" response, it would be overkill here as well. I will add a line about formatting the question to prompt the visitor to enter their email address

*****

Similar to email, this step type prompts the visitor to enter their phone number, which can be used
to follow up with additional information, schedule demos, and more.

.. warning::

>    Due to the vast number of formats used for phone numbers worldwide, responses to this step type

Reviewer: Does this only allow for numbers and special characters, like a +? If so, that could be mentioned.

Author: reponses are not validated, and can include special characters, numbers, and letters. I added a note to the warning to specify this.

*****

Similar to email, this step type prompts the visitor to enter their phone number, which can be used
to follow up with additional information, schedule demos, and more.

.. warning::
   Due to the vast number of formats used for phone numbers worldwide, responses to this step type
   are **not** validated for formatting, and can include both numbers and special characters.

Forward to Operator
*******************

This step forwards the conversation to an active live chat operator, so they can continue
assisting the visitor. As the conversation transcript is passed on to the operator, they can pick up
where the chatbot left off. This not only saves time for all parties involved, it can also help
qualify conversations before they reach live operators.

.. note::
   If no active operator is available on the channel, the chatbot continues the conversation with
   the visitor. Therefore, additional steps should be added after this one to ensure there is no
   abrupt end to the conversation. The additional steps can both inform visitors about the lack of

>    available operators (e.g.- `Uh-oh, it looks like none of our operators are available`) and
>    continue the conversation (e.g. - `Would you like to leave your email address?`).

Reviewer 2: These hyphens seem unnecessary, if you leave them in please remove the space before the second one :)
```suggestion
   available operators (e.g. `Uh-oh, it looks like none of our operators are available`) and
   continue the conversation (e.g. `Would you like to leave your email address?`).
```

*******************

This step forwards the conversation to an active live chat operator, so that they can continue
assisting the visitor. As the conversation transcript is passed on to the operator, they can pick up
where the chatbot left off. This not only saves time for all parties involved, it can also help
qualify conversations before they reach live operators.

.. note::

>    If no active operator is available on the channel, the chatbot continues the conversation with

Reviewer: Just a suggestion- but you could add the message from the image in this note. Because when I read 'additional steps' i thought of more questions and answers, not an 'everyone is busy'. Again, not necessary, just a suggestion.

Free Input/Multi-Line
*********************

The *free input* step allows visitors to respond to questions without providing pre-written
responses. Information provided in these responses is stored in the chat transcripts.

Choose between :guilabel:`Free Input` and :guilabel:`Free Input (Multi-Line)` depending on the type
and amount of information the visitor is asked to provide.

Create Lead
***********

> This step creates a lead in the *CRM* application. Select an option from the :guilabel:`Sales Team`

Reviewer: Totally optional, but you could use the abbreviation markup to spell out what CRM means/stands for.

drop-down to assign the created lead to a specific team.

.. note::
   This step is **only** available if the *CRM* application is installed on the database.

Create Ticket
*************

This step creates a :doc:`ticket </applications/services/helpdesk/overview/receiving_tickets>` in
the *Helpdesk* application. Select an option from the :guilabel:`Helpdesk Team` drop-down to assign
the created ticket to a specific team.

.. note::
   This step is **only** available if the *Helpdesk* application is installed on the database.

Only if
~~~~~~~

Chatbot scripts operate on an if/then basis, which means the next question presented to the visitor
is determined by the answer provided to the previous question.

> To continue the progression of the conversation, the form for a new step contains a field labeled

Reviewer: ```suggestion
To continue the progression of the conversation, the :guilabel:`Create Script Steps` form for a new step contains a field labeled
```

Create Lead
***********

This step creates a lead in the *CRM* application. Select an option from the :guilabel:`Sales Team`
drop-down to assign the created lead to a specific team.

.. note::
   This step is **only** available if the *CRM* application is installed on the database.

Create Ticket
*************

This step creates a :doc:`ticket </applications/services/helpdesk/overview/receiving_tickets>` in
the *Helpdesk* application. Select an option from the :guilabel:`Helpdesk Team` drop-down to assign
the created ticket to a specific team.

.. note::
   This step is **only** available if the *Helpdesk* application is installed on the database.

Only if
~~~~~~~

Chatbot scripts operate on an if/then basis, which means the next question presented to the visitor
is determined by the answer provided to the previous question.

To continue the progression of the conversation, the :guilabel:`Create Script Steps` form for a new
step contains a field labeled :guilabel:`Only If`. This field is where the progression of questions
is defined.

If a step is intended to follow all of the previous messages, this field can be left empty. However,

> if a message should **only** be sent conditionally, based on a previous response or several previous

Reviewer 3: ```suggestion
if a message should **only** be sent conditionally, based on a previous response, or several previous
```

> responses, those responses need to be added to this field.

Reviewer 3: ```suggestion
responses, those responses **must** be added to this field.
```

.. important::
   If any selections are made in the :guilabel:`Only If` field, they must **all** be selected during

>    the conversation before this step will be included. Only include selections in this field if they

Reviewer 3: ```suggestion
   the conversation *before* this step will be included. Only include selections in this field if they
```

Free Input/Multi-Line
*********************

The *free input* step allows visitors to respond to questions without providing pre-written
responses. Information provided in these responses is stored in the chat transcripts.

Choose between :guilabel:`Free Input` and :guilabel:`Free Input (Multi-Line)` depending on the type
and amount of information the visitor is asked to provide.

Create Lead
***********

This step creates a lead in the *CRM* application. Select an option from the :guilabel:`Sales Team`
drop-down field that appears to assign the created lead to a specific team.

.. note::
   This step is **only** available if the *CRM* application is installed on the database.

Create Ticket
*************

> This step creates a :doc:`ticket </applications/services/helpdesk/overview/receiving_tickets>` in

Reviewer 2: ```suggestion
This step creates a :doc:`ticket <../../services/helpdesk/overview/receiving_tickets>` in
```

Script testing
--------------

To ensure all visitors have a satisfactory experience with the chatbot, each message needs
to lead to a natural conclusion. Chatbot scripts should be tested to confirm there are no dead-ends,
and to understand what the visitor sees when they interact with the chatbot.

.. important::

>    If the visitor provides an answer or input that is **not** assigned a corresponding follow-up

Reviewer 3: ```suggestion
   If the visitor provides an answer, or input, that is **not** assigned a corresponding follow-up
```

>    response, the conversation stops (*dead-ends*). Since the visitor cannot re-engage the chatbot,

Reviewer 3: ```suggestion
   response, the conversation stops (reaches a *dead-end*). Since the visitor cannot re-engage the chatbot,
```

After a chatbot has been created and tested, it needs to be added to a live chat channel.

> First, open the :menuselection:`Live Chat` application, select the Kanban card for a

Reviewer: This is a touch confusing because you don't select the card- you hover over it to reveal the 3 dots, then click the dots. Just a suggestion to reword this.

:guilabel:`Channel`, and click the :guilabel:`⋮ (three dots)` icon to open the drop-down menu. Click
:guilabel:`Configure Channel` to open the channel detail form.

.. note::
   To create a new live chat channel, open the :menuselection:`Live Chat app` and click

>    :guilabel:`New`. Find more information :doc:`here </applications/websites/livechat>`.

Reviewer: I believe we want to use the doc name instead of a generic 'here'?

After a chatbot has been created and tested, it needs to be added to a live chat channel.

First, open the :menuselection:`Live Chat` application, find the Kanban card for the appropriate
live chat channel, hover over it, and click the :guilabel:`⋮ (three dots)` icon to open the
drop-down menu. Click :guilabel:`Configure Channel` to open the channel detail form.

.. note::
   To create a new live chat channel, open the :menuselection:`Live Chat app` and click

>    :guilabel:`New`. See :doc:`Live Chat </applications/websites/livechat>` for more information.

Reviewer 2: ```suggestion
   :guilabel:`New`. See :doc:`Live Chat <../livechat>` for more information.
```

Click on the :guilabel:`Channel Rules` tab. Then, open an existing rule, or create a new one by
clicking :guilabel:`Add a line`.

> On the :guilabel:`Create Rules` pop-up detail form, choose the appropriate chatbot in the
> :guilabel:`Chatbot` field.

Reviewer 2: ```suggestion
On the :guilabel:`Create Rules` modal, choose the appropriate chatbot in the :guilabel:`Chatbot`
field.
```
